extensions/liquibase-mongodb/deployment/src/main/java/io/quarkus/liquibase/mongodb/deployment/LiquibaseProcessor.java

```diff
@@ -0,0 +1,331 @@
+import io.quarkus.deployment.annotations.Consume;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.IndexDependencyBuildItem;
+import io.quarkus.deployment.builditem.ServiceStartBuildItem;
+import io.quarkus.deployment.builditem.SystemPropertyBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.NativeImageResourceBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.NativeImageResourceBundleBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.RuntimeInitializedClassBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ServiceProviderBuildItem;
+import io.quarkus.deployment.pkg.steps.NativeOrNativeSourcesBuild;
+import io.quarkus.deployment.util.ServiceUtil;
+import io.quarkus.liquibase.LiquibaseMongodbFactory;
+import io.quarkus.liquibase.runtime.LiquibaseMongodbBuildTimeConfig;
+import io.quarkus.liquibase.runtime.LiquibaseMongodbConfig;
+import io.quarkus.liquibase.runtime.LiquibaseMongodbRecorder;
+import io.quarkus.mongodb.runtime.MongodbConfig;
+import liquibase.change.Change;
+import liquibase.change.DatabaseChangeProperty;
+import liquibase.change.core.CreateProcedureChange;
+import liquibase.change.core.CreateViewChange;
+import liquibase.change.core.LoadDataChange;
+import liquibase.change.core.SQLFileChange;
+import liquibase.changelog.ChangeLogParameters;
+import liquibase.changelog.ChangeSet;
+import liquibase.changelog.DatabaseChangeLog;
+import liquibase.exception.LiquibaseException;
+import liquibase.parser.ChangeLogParser;
+import liquibase.parser.ChangeLogParserFactory;
+import liquibase.resource.ClassLoaderResourceAccessor;
+
+class LiquibaseProcessor {
+
+    private static final Logger LOGGER = Logger.getLogger(LiquibaseProcessor.class);
+
+    private static final String LIQUIBASE_BEAN_NAME_PREFIX = "liquibase_";
+
+    private static final DotName DATABASE_CHANGE_PROPERTY = DotName.createSimple(DatabaseChangeProperty.class.getName());
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(Feature.LIQUIBASE_MONGODB);
+    }
+
+    @BuildStep
+    public SystemPropertyBuildItem disableHub() {
+        // Don't block app startup with prompt:
+        // Do you want to see this operation's report in Liquibase Hub, which improves team collaboration?
+        // If so, enter your email. If not, enter [N] to no longer be prompted, or [S] to skip for now, but ask again next time (default "S"):
+        return new SystemPropertyBuildItem("liquibase.hub.mode", "off");
+    }
+
+    @BuildStep(onlyIf = NativeOrNativeSourcesBuild.class)
+    IndexDependencyBuildItem indexLiquibase() {
+        return new IndexDependencyBuildItem("org.liquibase", "liquibase-core");
```

Comment:
What do we need from `liquibase-core` that mandates an indexing of it?

Reply:
I don't really know, I copied this step from the liquibase extension.
I can try without it and see if it stil works in native.

Reply:
Yeah, if we can get away without it, cool. If not, no problem

Reply:
It's not needed, I didn't test Java migrations thought, but currently without it the IT pass.
As the current Liquibase extension didn't test Java miugrations either I don't know if they pass on native.